### ISSUE TITLE
C++ ModuleDict

### ISSUE THREAD

**Issue by meganset** [pytorch/pytorch#45896]
Is ModuleDict still slated to be added to the C++ api?

It looks like pr #28652, opened in Oct 2019 was close to completion,
then was never finished.

We can take the code in this pr's _moduledict.h_ for now,
but wanted to check if the container is still destined for the C++ api at some point.

Thanks.

**Comment by albanD** [comment]
See https://github.com/pytorch/pytorch/issues/25883 for more details on the API parity work.

**Comment by glaringlee** [comment]
@meganset 
Yes, we will land this into pytorch code base. The author seems no longer working on it, will double check with the author and find someone else wrapping this up if necessary.
